packages/textkit/src/engines/linebreaker/index.js

```diff
@@ -17,6 +18,50 @@ const opts = {
   shrink: 9,
 };
 
+/**
+ * Get regions of urls in the attributed string
+ *
+ * @param {Object} attributedString attributed string
+ * @returns {Array} regions of urls
+ */
+const getUrlRegions = attributedString => {
+  const urlRegions = [];
+
+  const words = attributedString.string.split(/([ ]+)/g).filter(Boolean);
+  let start = 0;
+  for (let index = 0; index < words.length; index += 1) {
+    const word = words[index];
+    if (isUrl(word)) {
+      urlRegions.push({ start, end: start + word.length });
```

Comment:
You've missed to count spaces for the ranges

Reply:
No. The string will split with the spaces
![Screenshot 2022-04-22 at 9 11 33 AM](https://user-images.githubusercontent.com/31011971/164591957-297f79ff-ce9d-4fef-b0c3-9303d3b0f89d.png)